### PR TITLE
feat: add UEN field

### LINKED ISSUE #1273: UEN field
- Helps to reduce typos in UENs and ensure data accuracy

https://gist.github.com/mervintankw/90d5660c6ab03a83ddf77fa8199a0e52

### PR DESCRIPTION
**Pull request by aniruddha-adhikary**
## Problem
It adds a UEN Field

Closes #1273 

## Solution
I added a new field in the backend, frontend admin and public frontend.

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
    - Details ...
- [x] No - this PR is backwards compatible  

**Features**:
- New Field for UEN
- Validates UEN number syntax/format

## Before & After Screenshots

**BEFORE**:
[Imagine no UEN Fields]

**AFTER**:
Screenshot 2021-06-08 at 2 00 56 PM
Screenshot 2021-06-08 at 2 02 07 PM

## Deploy Notes
I am not sure if adding a new field would require any sort of migration. Maintainers, please do edit if you think it needs to.

### CODE REVIEW

File: src/app/utils/field-validation/answerValidator.factory.ts [hunk 1]
```diff
@@ -76,6 +78,8 @@ export const constructSingleAnswerValidator = (
     return constructDropdownValidator(formField)
   } else if (isEmailField(formField)) {
     return constructEmailValidator(formField)
+  } else if (isUenValid(formField)) {
```
Comment (mantariksh) [inline src/app/utils/field-validation/answerValidator.factory.ts:81]: this should be `isUenField` right?

**Review by mantariksh — COMMENTED** [review]
not sure why the CICD is failing, will check it out

File: src/app/utils/field-validation/answerValidator.factory.ts (hunk 1 above)
Comment (aniruddha-adhikary) [reply src/app/utils/field-validation/answerValidator.factory.ts:81]: Ah good catch, yes! Let me fix that.

Comment (mantariksh) [reply src/app/utils/field-validation/answerValidator.factory.ts:81]: actually I believe all the tests will pass once you fix this

**Comment by mantariksh** [comment]
not sure why our github actions for frontend and backend tests aren't running on this PR. I'll look into our github actions config to fix this. shouldn't be a big deal since I've run the tests locally and everything checks out.

File: src/app/utils/field-validation/answerValidator.factory.ts (hunk 1 above)
Comment (aniruddha-adhikary) [reply src/app/utils/field-validation/answerValidator.factory.ts]: I've pushed out the update man.

**Comment by mantariksh** [comment]
@aniruddha-adhikary this lgtm but unfortunately we're not able to run the right github actions on this PR. to expedite this, I'm going to do the following:
1. pull your commits
2. push the commits to a branch in this repo and open a new PR
3. deploy this to our staging environment and get the rest of the team to take a look to make sure it's okay

hopefully we can get this into production during our release tomorrow, if not next week. will keep you updated. thanks again for the contribution!

**Comment by mantariksh** [comment]
## Manual tests
Create a UEN field and check that the following UENs are **valid**:
- [ ] 01234567A
- [ ] 012345678A
- [ ] T01LP1234Z
- [ ] S01LP1234Z
- [ ] R01LP1234Z

Check that the following UENs are **invalid**:
- [ ] B1234567A
- [ ] 012345678
- [ ] 0123456789
- [ ] A01LP1234Z
- [ ] S01AA1234Z
- [ ] RS1LP1234Z
- [ ] R010P1234Z
- [ ] R01LPL234Z
- [ ] R01LP12345

Submit the following UEN **by API** and check that the backend invalidates it:
- [ ] R01LP12345

**Review by mantariksh — APPROVED** [review]

**Comment by mantariksh** [comment]
@aniruddha-adhikary this went into production today, congrats on your first FormSG feature!

**Comment by aniruddha-adhikary** [comment]
@mantariksh 🥺 Thanks ya'all. You guys are excellent, responsible and responsive open source maintainers. ❤️ 🚀

**Pull request merged**